### PR TITLE
Frog and Snail Mutation Trees

### LINKED ISSUE #51569: Suggestion: Frog Mutation
# Describe the solution you'd like
With all the new mutation trees, I think that a frog mutation would be cool to see.

# Describe alternatives you've considered
Adding it myself ( no idea how to add mutations)
Not submitting this issue

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Content "Adds Frog and Snail Mutation Trees"

#### Purpose of change
Adds a few new mutation trees. Fixes #51569 .

#### Describe the solution
Adds mutations, thresholds, mutagen, recipes, spawns, and unique mutagen ingredients.

#### Describe alternatives you've considered
None

#### Testing
Ongoing

### CODE REVIEW

**Comment by actual-nh** [comment]
Ping: @Theawesomeboophis?

File: data/json/monsters/nether.json [hunk 1]
```diff
@@ -565,7 +565,7 @@
     "armor_bash": 6,
     "armor_cut": 12,
     "armor_bullet": 10,
-    "harvest": "demihuman",
+    "harvest": "demihuman_snail",
     "special_attacks": [ [ "ACID", 15 ] ],
     "anger_triggers": [ "PLAYER_WEAK", "FRIEND_DIED" ],
     "flags": [ "SMELLS", "HEARS", "POISON", "ACIDPROOF", "ACID_BLOOD" ]
```
Comment (actual-nh) [inline data/json/monsters/nether.json:571]: I just realized - apparently the eyestalks don't have functional eyes?

File: data/json/mutations/mutation_category.json [hunk 2]
```diff
@@ -66,6 +66,15 @@
     "iv_message": "Your pulse pounds as the waves.",
     "memorial_message": "Went deep."
   },
+  {
+    "type": "mutation_category",
+    "id": "GASTROPOD",
+    "name": "Gastropod",
+    "threshold_mut": "THRESH_GASTROPOD",
+    "mutagen_message": "Are you dancing a descending spiral?",
+    "iv_message": "The gaps between your body shrink.",
```
Comment (actual-nh) [inline data/json/mutations/mutation_category.json:75]: ?

File: data/json/mutations/mutation_spells.json [hunk 3]
```diff
@@ -19,5 +19,65 @@
...
+    "description": "Spray slime onto an enemy to slow them down.",
+    "shape": "blast",
+    "effect": "mod_moves",
+    "valid_targets": [ "hostile" ],
+    "min_damage": -90,
+    "max_damage": -335,
+    "min_range": 1,
+    "max_range": 1,
+    "spell_class": "NONE",
+    "flags": [ "SILENT", "NO_LEGS", "NO_HANDS", "RANDOM_DAMAGE", "RANDOM_DURATION", "RANDOM_TARGET" ],
```
Comment (actual-nh) [inline data/json/mutations/mutation_spells.json:36]: RANDOM_TARGET?

File: data/json/mutations/mutation_spells.json [hunk 4]
```diff
@@ -19,5 +19,65 @@
...
+    "max_level": 1,
+    "energy_source": "STAMINA"
+  },
+  {
+    "type": "SPELL",
+    "id": "spell_crushing_leap",
+    "name": "Crushing Leap",
+    "description": "You squat down, build up tension in your legs and release.  You have the ability to target and land in the way to do maximum damage to any target in your way.  Death from above.",
+    "valid_targets": [ "hostile", "ground" ],
+    "flags": [ "SILENT", "NO_LEGS", "NO_HANDS", "RANDOM_DAMAGE", "RANDOM_TARGET" ],
```
Comment (actual-nh) [inline data/json/mutations/mutation_spells.json:67]: RANDOM_TARGET?

File: data/json/mutations/mutations.json [hunk 5]
```diff
@@ -470,6 +491,19 @@
     "//": "nope.  This does NOT lead to EATPOISON.  Stomach problems are part of the GI upgrades--one advantage to not having Robust Genetics.",
     "cancels": [ "WEAKSTOMACH" ]
   },
+  {
+    "type": "mutation",
+    "id": "VOMITSTOMACH",
+    "name": { "str": "Stomach Cleanse" },
+    "points": 1,
+    "description": "You have gained the ability to invert your stomach outside your body to quickly eliminate toxins, and are more likely to throw up from food poisoning, alcohol, etc.  If you throw up because of this this, you won't suffer any residual nausea.",
+    "starting_trait": true,
```
Comment (actual-nh) [inline data/json/mutations/mutations.json:500]:
```suggestion
    "starting_trait": false,
```
I would think?

File: data/json/mutations/mutations.json [hunk 6]
```diff
@@ -1610,7 +1646,7 @@
     "id": "URSINE_EYE",
     "name": { "str": "Ursine Vision" },
     "points": 1,
-    "description": "Your visual processing has shifted: though you can see better in the dark, you're nearsighted in the light.  Maybe glasses would help.  Activate to toggle NV-visible areas on or off.",
+    "description": "Your visual processing has shifted: though you can see better in the dark, you're nearsighted in the light.  Maybe glasses would help.  Activate to toggle NV-visible areas on or off.  Surprisingly, frogs vision works exactly the same.",
```
Comment (actual-nh) [inline data/json/mutations/mutations.json:1649]:
```suggestion
    "description": "Your visual processing has shifted: though you can see better in the dark, you're nearsighted in the light.  Maybe glasses would help.  Activate to toggle NV-visible areas on or off.  Surprisingly, frogs' vision works exactly the same.",
```

File: data/json/mutations/mutations.json [hunk 7]
```diff
@@ -1831,7 +1867,7 @@
     "visibility": 5,
     "ugliness": 3,
     "description": "You've grown a set of gills in your neck, allowing you to breathe underwater.  Slightly increases wet benefits.",
-    "category": [ "FISH" ],
+    "category": [ "FISH", "BATRACHIAN" ],
```
Comment (actual-nh) [inline data/json/mutations/mutations.json:1870]: From the tadpole stage?

File: data/json/mutations/mutations.json [hunk 8]
```diff
@@ -3131,6 +3189,54 @@
...
+      "strength_damage": { "damage_type": "stab", "amount": 2 }
+    }
+  },
+  {
+    "type": "mutation",
+    "id": "GASTROPOD_EXTREMITY2",
+    "name": { "str": "Harpoon Tongue" },
+    "visibility": 5,
+    "ugliness": 5,
+    "description": "You've tongue has increased it's length and mobility.",
```
Comment (actual-nh) [inline data/json/mutations/mutations.json:3228]:
```suggestion
    "description": "Your tongue has increased its length and mobility.",
```

File: data/json/mutations/mutations.json [hunk 9]
```diff
@@ -3131,6 +3189,54 @@
...
+    "threshreq": [ "THRESH_GASTROPOD" ],
+    "category": [ "GASTROPOD" ],
+    "attacks": {
+      "attack_text_u": "You tear into %s with your harpoon tongue.",
+      "attack_text_npc": "%1$s tears into %2$s with their harpoon tongue.",
```
Comment (actual-nh) [inline data/json/mutations/mutations.json:3233]:
```suggestion
      "attack_text_u": "You stab into %s with your harpoon tongue.",
      "attack_text_npc": "%1$s stabs into %2$s with their harpoon tongue.",
```

File: data/json/mutations/mutations.json [hunk 10]
```diff
@@ -3131,6 +3189,54 @@
...
+    "threshreq": [ "THRESH_GASTROPOD" ],
+    "category": [ "GASTROPOD" ],
+    "cancels": [ "FANGS", "MANDIBLES" ],
+    "attacks": {
+      "attack_text_u": "You tear into %s with your rasping tongue.",
+      "attack_text_npc": "%1$s tears into %2$s with their rasping tongue.",
+      "body_part": "mouth",
+      "chance": 15,
+      "base_damage": { "damage_type": "stab", "amount": 15 },
+      "strength_damage": { "damage_type": "stab", "amount": 2 }
```
Comment (actual-nh) [inline data/json/mutations/mutations.json:3219]:
```suggestion
      "base_damage": { "damage_type": "cut", "amount": 15 },
      "strength_damage": { "damage_type": "cut", "amount": 2 }
```

File: data/json/mutations/mutations.json [hunk 11]
```diff
@@ -3153,6 +3259,45 @@
...
+  },
+  {
+    "type": "mutation",
+    "id": "LEAPING_LEGS2",
+    "name": { "str": "Crushing Leap" },
+    "points": 1,
+    "visibility": 3,
+    "ugliness": 1,
+    "types": [ "LEGS" ],
+    "description": "You've gained a strong ability to leap from a standing position a significant distance and to crush whatever you land on.",
```
Comment (actual-nh) [inline data/json/mutations/mutations.json:3296]:
```suggestion
    "description": "You've gained a strong ability to leap a significant distance from a standing position and crush whatever you land on.",
```

File: data/json/mutations/mutations.json [hunk 12]
```diff
@@ -4746,6 +4969,35 @@
...
+    "changes_to": [ "LONG_TONGUE2" ],
+    "description": "Your tongue has lengthened and often extends far beyond your mouth.  This has no effect.",
+    "category": [ "BATRACHIAN" ]
+  },
+  {
+    "type": "mutation",
+    "id": "LONG_TONGUE2",
+    "name": { "str": "Whipping Tongue" },
+    "visibility": 1,
+    "description": "You've developed an extremely long and mobile tongue.  You may use it to attack your enemies.",
```
Comment (actual-nh) [inline data/json/mutations/mutations.json:4988]: Perhaps a spell for a longer-ranged and sticky version of this?

File: data/json/mutations/mutations.json [hunk 13]
```diff
@@ -4844,7 +5113,7 @@
     "scent_intensity": 1200,
     "prereqs": [ "SMELLY" ],
     "leads_to": [ "PHEROMONE_INSECT", "PHEROMONE_MAMMAL" ],
```
Comment (actual-nh) [inline data/json/mutations/mutations.json:5115]:
```suggestion
    "leads_to": [ "PHEROMONE_INSECT", "PHEROMONE_MAMMAL", "PHEROMONE_AMPHIBIAN" ],
```

File: data/json/mutations/mutations.json [hunk 14]
```diff
@@ -6180,15 +6474,35 @@
...
+    "type": "mutation",
+    "id": "SHELL3",
+    "name": { "str": "Nacreous Shell" },
+    "points": -1,
+    "visibility": 10,
+    "ugliness": -2,
+    "mixed_effect": true,
+    "description": "Your protective shell has developed increased strength and hardness and a mother of pearl sheen.",
+    "encumbrance_always": [ [ "torso", 15 ] ],
+    "prereqs": [ "SHELL" ],
```
Comment (actual-nh) [inline data/json/mutations/mutations.json:6497]:
```suggestion
    "prereqs": [ "SHELL2" ],
```

File: data/json/mutations/mutations.json [hunk 15]
```diff
@@ -6235,6 +6549,16 @@
     "purifiable": false,
     "threshold": true
   },
+  {
+    "type": "mutation",
+    "id": "THRESH_GASTROPOD",
+    "name": { "str": "Gastropod" },
+    "points": 1,
+    "description": "You feel bad for all the creatures that don't carry their home on their back.",
```
Comment (actual-nh) [inline data/json/mutations/mutations.json:6557]:
```suggestion
    "description": "You feel bad for all the creatures that don't carry their home on their backs.",
```

**Review by actual-nh — COMMENTED** [review]
I like!

File: data/json/mutations/mutation_category.json (hunk 2 above)
Comment (Maleclypse) [reply data/json/mutations/mutation_category.json:75]: I'm open to suggestions.  the first I was making an oblique reference to spiral shells and the second to the PCs legs slowly merging into a single large "foot"

File: data/json/monsters/nether.json (hunk 1 above)
Comment (Maleclypse) [reply data/json/monsters/nether.json:571]: In game or irl?

File: data/json/mutations/mutation_spells.json (hunk 3 above)
Comment (Maleclypse) [reply data/json/mutations/mutation_spells.json:36]: I thought random target means it randomly targets a body part.

File: data/json/mutations/mutations.json (hunk 7 above)
Comment (Maleclypse) [reply data/json/mutations/mutations.json:1870]: I think there might be a different trait that would make this superfluous. But I'd need to look for it specifically.

File: data/json/monsters/nether.json (hunk 1 above)
Comment (actual-nh) [reply data/json/monsters/nether.json:571]: IRL, they do. This critter has neither vision ranges specified nor SEES as a flag?

File: data/json/mutations/mutation_category.json (hunk 2 above)
Comment (actual-nh) [reply data/json/mutations/mutation_category.json:75]: The first one definitely works. I'll have to think about the second one.

File: data/json/mutations/mutation_spells.json (hunk 3 above)
Comment (actual-nh) [reply data/json/mutations/mutation_spells.json:36]:
```suggestion
    "flags": [ "SILENT", "NO_LEGS", "NO_HANDS", "RANDOM_DAMAGE", "RANDOM_DURATION" ],
```
No - "random valid target within range" according to doc/MAGIC.md.

File: data/json/monsters/nether.json (hunk 1 above)
Comment (Maleclypse) [reply data/json/monsters/nether.json:571]: That probably needs a fix.

File: data/json/mutations/mutations.json (hunk 12 above)
Comment (Maleclypse) [reply data/json/mutations/mutations.json:4988]: I wasn't able to figure out the C++ needed to make it sticky to pull a monster in to you.   So I haven't done the spell version.

File: data/json/monsters/nether.json (hunk 1 above)
Comment (actual-nh) [inline data/json/monsters/nether.json:571]:
```suggestion
    "flags": [ "SEES", "SMELLS", "HEARS", "POISON", "ACIDPROOF", "ACID_BLOOD" ]
```

File: src/map_field.cpp [hunk 16]
```diff
@@ -1468,7 +1469,7 @@ void map::player_in_field( Character &you )
         }
         if( ft == fd_sludge ) {
             // Sludge is on the ground, but you are above the ground when boarded on a vehicle
-            if( !you.in_vehicle ) {
+            if( !u.has_trait( trait_GASTROPOD_FOOT ) && ( !u.in_vehicle ) ) {
```
Comment (Maleclypse) [inline src/map_field.cpp:1472]:
```suggestion
            if( !you.has_trait( trait_GASTROPOD_FOOT ) && ( !you.in_vehicle ) ) {
```

File: data/json/mutations/mutation_category.json (hunk 2 above)
Comment (actual-nh) [reply data/json/mutations/mutation_category.json:75]:
```suggestion
    "iv_message": "Your split lower body starts to come back together.",
```

**Review by actual-nh — COMMENTED** [review]
Evidently the name of the mutation/whatever is used in naming inline enchantments?

File: data/json/mutations/mutations.json [hunk 17]
```diff
@@ -307,6 +307,27 @@
     "category": [ "MOUSE", "RABBIT" ],
     "enchantments": [ { "condition": "ALWAYS", "values": [ { "value": "SLEEPY", "add": 40 } ] } ]
   },
+  {
+    "type": "mutation",
+    "id": "DIRTSLEEPER",
+    "name": { "str": "Practiced Sleeper" },
```
[the remark this answers is not in the document]
Comment (Maleclypse) [reply data/json/mutations/mutations.json:313]: I feel like that's a very polite way to describe my method,  if I find a mutation close to what I want to do I copy/paste then change as few things as possible to accomplish the effect I'm looking for.  I forgot to change the name in these two.

**Comment by Venera3** [comment]
Cool stuff! You could add some threshold dreams, and I could see an intermediate Eyestalk level where you're disoriented by the newfound mobility of your eyes - the visual malfunction bionic got moved over to an EOC so that should be doable.

File: data/json/mutations/mutations.json (hunk 7 above)
Comment (Maleclypse) [reply data/json/mutations/mutations.json:1870]: OK so I'm thinking I see what you mean, I should ditch the gills and create a unique mutation that has the same impact as gills but comes from the osmatic skin type stuff?

Comment (Maleclypse) [reply data/json/mutations/mutations.json:1870]: Ah here it is  Gills is a prerequisite to getting  
``` {
    "type": "mutation",
    "id": "FRESHWATEROSMOSIS",
    "name": { "str": "Freshwater Osmosis" },
    "points": 4,
    "description": "Your gills and skin are highly permeable to moisture, and you can hydrate your body through osmosis.  While submerged in fresh water, any thirst you might have will be slaked.",
    "prereqs": [ "GILLS" ],
    "category": [ "FISH", "BATRACHIAN" ],
    "threshreq": [ "THRESH_FISH", "THRESH_BATRACHIAN" ]
  },
```

**Comment by actual-nh** [comment]
@Maleclypse: You have some conflicts needing resolving.

**Comment by Maleclypse** [comment]
> @Maleclypse: You have some conflicts needing resolving.

Which is weird. Because I can't see them.  And they are probably mine from another PR.

**Comment by Maleclypse** [comment]
maleclypse — Today at 10:00 PM
```        for( const std::string &elem : it->conditions ) {
            //Collect valid time of day ranges
            if( elem == "DAY" || elem == "NIGHT" || elem == "DUSK" || elem == "DAWN" ) {
                if( elem == "DAY" ) {
                    valid_times_of_day.emplace_back( sunrise, sunset );
                } else if( elem == "NIGHT" ) {
                    valid_times_of_day.emplace_back( sunset, sunrise );
                } else if( elem == "DUSK" ) {
                    valid_times_of_day.emplace_back( sunset - 1_hours, sunset + 1_hours );
                } else if( elem == "DAWN" ) {
                    valid_times_of_day.emplace_back( sunrise - 1_hours, sunrise + 1_hours );
                }
            }
```

I want to add Night and Day as active conditions in enchantments   
`if( active_conditions.second == condition::NIGHT ) {
        return guy.pos().z < 0;
    }`
   I get that I need to change the return guy.pos().z < 0;  but I'm not sure how to pull into the day ranges that I think are elem

maleclypse — Today at 11:01 PM
```
Is it     if( active_conditions.second == condition::NIGHT ) {
        return is_night;
    }```  
?
and to get that to work do I need to include the calendar.h?

**Comment by Maleclypse** [comment]
failed tests


These tests are failing but when I look at the tests folder I can't see why they would be failing now.

**Comment by Maleclypse** [comment]
Well now I have to figure out how the points are awarded to a given mutation.   Ugly somehow awards 24 Batrachian points by itself.

File: data/json/dreams.json [hunk 18]
```diff
@@ -326,6 +377,15 @@
     "category": "LIZARD",
     "strength": 3
   },
+  {
+    "type": "dream",
+    "messages": [
+      "You have a strange dream about being the Easter Bunny.",
+      "You have a comforting and warm dream of living as a rabbit."
+    ],
+    "category": "LIZARD",
```
Comment (Termineitor244) [inline data/json/dreams.json:386]: Shouldn't this be "RABBIT"?

File: data/json/items/comestibles/mutagen.json [hunk 19]
```diff
@@ -225,6 +225,24 @@
...
+    "description": "A super-concentrated mutagen with the consistency of snot.  You need a syringe to inject it… if you really want to?",
+    "color": "light_gray",
+    "use_action": { "type": "mutagen_iv", "mutation_category": "GASTROPOD" }
+  },
+  {
+    "id": "iv_mutagen_batrachian",
+    "copy-from": "iv_mutagen_flavor",
+    "type": "COMESTIBLE",
+    "name": { "str": "batrachian serum" },
+    "description": "A super-concentrated mutagen with that looks like pond water… are the shadows in it moving?  You need a syringe to inject it… if you really want to?",
```
Comment (Termineitor244) [inline data/json/items/comestibles/mutagen.json:242]: "A super-concentrated mutagen **with that looks**"
I think the "with" should be removed... Or maybe I'm just bad at English?

File: data/json/items/comestibles/mutagen.json [hunk 20]
```diff
@@ -409,6 +427,22 @@
     "name": { "str": "spider mutagen" },
     "use_action": { "type": "mutagen", "mutation_category": "SPIDER" }
   },
+  {
+    "id": "mutagen_batrachian",
+    "copy-from": "mutagen_flavor",
+    "type": "COMESTIBLE",
+    "name": { "str": "batrachian mutagen" },
+    "description": "A shadowy mutagen that looks like might be full of tiny swimming tadpoles.",
```
Comment (Termineitor244) [inline data/json/items/comestibles/mutagen.json:435]: "A shadowy mutagen that **looks like might**"
Like "**it**" might?

File: data/json/mutations/mutations.json [hunk 21]
```diff
@@ -245,6 +258,34 @@
...
+    "enchantments": [
+      { "condition": "UNDERGROUND", "values": [ { "value": "BONUS_BLOCK", "add": 1 }, { "value": "BONUS_DODGE", "add": 1 } ] }
+    ]
+  },
+  {
+    "type": "mutation",
+    "id": "NIGHT_RAIDER",
+    "name": { "str": "Night Raider" },
+    "points": 3,
+    "description": "You know how to fight in cramped tunnels at risk of collapse.  You gain bonuses when fighting underground.",
```
Comment (Termineitor244) [inline data/json/mutations/mutations.json:278]: Is this supposed to have the same description as "TUNNEL_FIGHTER"?

File: data/json/mutations/mutations.json [hunk 22]
```diff
@@ -568,7 +670,7 @@
     "id": "TERRIFYING",
     "name": { "str": "Terrifying" },
     "points": 1,
-    "description": "There's something about you that creatures find frightening, and they are more likely to try to flee.",
+    "description": "There's something about you that humans and post-humans find frightening, and they are more likely to try to flee.",
```
Comment (Termineitor244) [inline data/json/mutations/mutations.json:673]: This affects the moral of all creatures:
```
if( u->has_trait( trait_TERRIFYING ) ) {
            effective_morale -= 10;
        }
```
(From src/monster.cpp)
So "creatures" is more appropriate.

File: data/json/items/comestibles/mutagen.json (hunk 19 above)
Comment (Zireael07) [reply data/json/items/comestibles/mutagen.json:242]: Seconding, the with should be removed.

File: data/json/dreams.json (hunk 18 above)
Comment (Maleclypse) [reply data/json/dreams.json:386]: Thanks! This was already merged so someone will have to create a new PR with the fixes.   Let me know if you want to that and if not I'll put them on my todo list.

Comment (Termineitor244) [reply data/json/dreams.json:386]: Oh, then I will do the PR, serves me as an excuse to update my master, just a question, NIGHT_RAIDER is supposed to have the same description as TUNNEL_FIGHTER? To know if I should change it too.

Comment (Maleclypse) [reply data/json/dreams.json:386]: no it should have different text .  I just forgot to update it.

Comment (Termineitor244) [reply data/json/dreams.json:386]: Done #51859.

**Pull request merged**